tests/js/common/shell/multi/shell-validation.js

```diff
@@ -708,6 +708,70 @@ function ValidationBasicsSuite () {
   }; // return
 } // END - ValidationBasicsSuite
 
+
+function UpdateSchemaCoverageSuite() {
+  const testCollectionName = "TestCollection";
+  let testCollection = null;
+  const validatorJson = {
+    "message": "",
+    "level": "new",
+    "type": "json",
+    "rule": {
+      "additionalProperties": true,
+      "properties": {
+        "created": {
+          "type": "integer"
+        },
+        "creator": {
+          "type": "string"
+        },
+        "name": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "created",
+        "creator"
+      ],
+      "type": "object"
+    }
+  };
+
+
+  return {
+
+    setUp: () => {
+      try {
+        db._drop(testCollectionName);
+      } catch (ex) {}
+      testCollection = db._create(testCollectionName, {
+        schema: validatorJson,
+        replicationFactor: 3,
+        numberOfShards: 5
+      });
+    },
+
+    tearDown: () => {
+      try {
+        db._drop(testCollectionName);
+      } catch (ex) {}
+    },
+
+    testPropertiesRemoveAttributeAfterInsertion: () => {
+      assertNotNull(testCollection);
+      const newDoc = {"created": 123, "creator": "Julia"};
+      testCollection.insert(newDoc);
+      delete validatorJson.rule.properties.name;
+      assertFalse(validatorJson.rule.hasOwnProperty("name"));
```

Comment:
you need to do this *after* editing the schema on the then server loaded schema.